src/app/idps/[idpId]/page.tsx

```diff
@@ -9,6 +12,14 @@ type IdpPageProps = {
 };
 
 export default async function Idp(props: Readonly<IdpPageProps>) {
+	const session = await auth.api.getSession({
```

Comment:
If you want you could implement something like the following in `@/lib/auth`

```ts
export type Session = typeof auth.$Infer.Session;

// server side auth session
export async function getSession(): Promise<Session | null> {
  return await auth.api.getSession({
    headers: await headers(),
  });
}
```

and `import { getSession } from '@/lib/auth';` in your server pages